### PR TITLE
[Docs]: Add Propagate Dependencies user guide

### PR DESCRIPTION
**Pull request by hanweisen**
Signed-off-by: hanweisen <hanweisen_yewu@cmss.chinamobile.com>

**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Add Propagate Dependencies user guide
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by hanweisen** [comment]
This feature only supports propagate ConfigMap and Secret , I'm curious why it is not supported Service and PVC.

**Comment by RainbowMango** [comment]
> This feature only supports propagate ConfigMap and Secret , I'm curious why it is not supported Service and PVC.

The `ConfigMap` and `Secret` are just a start because it's easier to implement. The door is open for other dependencies, we can iterate it one by one.

**Comment by RainbowMango** [comment]
cc @Tingtal Could you please help with it?
Looks pretty good to me.

File: docs/userguide/propagate-dependencies.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# Propagate dependencies
+Deployment, Job, Pod, DaemonSet and StatefulSet dependencies(ConfigMaps and Secrets) can be propagated to the member
```
Comment (Tingtal) [inline docs/userguide/propagate-dependencies.md:2]:
```suggestion
Deployment, Job, Pod, DaemonSet and StatefulSet dependencies (ConfigMaps and Secrets) can be propagated to member
```

File: docs/userguide/propagate-dependencies.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+clusters automatically. This document demonstrates how to use this feature. More design details please referring to
```
Comment (Tingtal) [inline docs/userguide/propagate-dependencies.md:3]:
```suggestion
clusters automatically. This document demonstrates how to use this feature. For more design details, please refer to
```

File: docs/userguide/propagate-dependencies.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+          - targetCluster:
+              clusterNames:
+                - member1
+            weight: 1
+          - targetCluster:
+              clusterNames:
+                - member2
+            weight: 1
+```
+After apply them. Deployment and ConfigMap are properly propagated to member cluster.
```
Comment (Tingtal) [inline docs/userguide/propagate-dependencies.md:91]:
```suggestion
Upon successful policy execution, the Deployment and ConfigMap are properly propagated to the member cluster.
```

**Review by Tingtal — CHANGES_REQUESTED** [review]

**Comment by Tingtal** [comment]
> cc @Tingtal Could you please help with it? Looks pretty good to me.

@RainbowMango Hi! Language checked. Just a few small touches.

**Comment by RainbowMango** [comment]
Please help squash the commits.

**Comment by hanweisen** [comment]
Thanks a lot @Tingtal. These commit has been squashed @RainbowMango

**Comment by hanweisen** [comment]
Thanks a lot @Tingtal. These commit has been squashed @RainbowMango

**Pull request merged**